CB-9589 auto convert IDs to new style IDs if in registry mapper

stevengill opened the pull request:
@nikhilkh @vladimir-kotikov take a look. Thoughts?

vladimir-kotikov commented on cordova-lib/src/plugman/install.js:
```diff
@@ -79,6 +80,19 @@ module.exports = function installPlugin(platform, project_dir, id, plugins_dir,
+    //if it does, convert id to new name id 
+    var newId = pluginMapper[splitVersion[0]];
+    if(newId) {
+        events.emit('log', 'Notice: ' + id + ' has been automatically converted to ' + newId + ' and fetched from npm. This is due to our old plugins registry shutting down.');
```
Shouldn't this be emitted with `warn` level to encourage users to switch to new plugin id instead of relying on plugin mapper?

vladimir-kotikov commented on cordova-lib/src/plugman/install.js:
```diff
@@ -404,6 +418,20 @@ function installDependencies(install, dependencies, options) {
+                //if it does, convert id to new name id 
+                var newId = pluginMapper[splitVersion[0]];
+                if(newId) {
+                    events.emit('log', 'Notice: ' + dep.id + ' has been automatically converted to ' + newId + ' and fetched from npm. This is due to our old plugins registry shutting down.');
```
Shouldn't this be emitted with `warn` level to encourage users to switch to new plugin id instead of relying on plugin mapper?

vladimir-kotikov commented on cordova-lib/src/plugman/uninstall.js:
```diff
@@ -142,8 +143,17 @@ module.exports.uninstallPlugin = function(id, plugins_dir, options) {
+            if (!fs.existsSync(deps_path)) {
+                var newId = pluginMapper[splitVersion[0]];
+                if (newId && toDelete.indexOf(newId) === -1) {
+                   toDelete.push(newId);
```
IMHO it makes sense to add verbose message here to show to user what's happened.

vladimir-kotikov commented on cordova-lib/src/plugman/uninstall.js:
```diff
@@ -256,6 +266,15 @@ function runUninstallPlatform(actions, platform, project_dir, plugin_dir, plugin
+                var splitVersion = dangler.split('@');
+                var newId = pluginMapper[splitVersion[0]];
+                if(newId) {
+                    dependent_path = path.join(plugins_dir, newId);
```
Same here.

stevengill commented:
@vladimir-kotikov switched to warnings and added some verbose output. Also added a conversion step to fetch.js as it was needed in there since cordova-lib calls fetch instead of install. I originally had my ID conversion code in registry.js, but this lead to some failing checks in the code for what ID was expected and what was actually retrieved. i think it is all good now. Let me know if you have other thoughts before I merge.

vladimir-kotikov commented:
@stevengill, looks good :+1:

stevengill commented:
merged
